acc/components/optics/guide_baseline.py

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/env python
+#
+# Copyright (c) 2021 by UT-Battelle, LLC.
+
+from math import inf, isnan, tanh, sqrt, ceil
+from numba import cuda
+import numpy as np
+
+from mcni.AbstractComponent import AbstractComponent
+from mcni.neutron_storage import neutrons_as_npyarr, ndblsperneutron
+
+category = 'optics'
+
+# mcni.utils.conversion.v2k
+from mcni.utils.conversion import V2K
+# In mcstas header
+# V2K = 1.58825361e-3
+@cuda.jit(device=True, inline=True)
+def v2k(v):
+    """v in m/s, k in inverse AA """
+    return v * V2K
+
+@cuda.jit(device=True, inline=True)
+def calc_reflectivity(Q, R0, Qc, alpha, m, W):
+    """
+    Calculate the mirror reflectivity for a neutron.
+
+    Returns:
+    float: the reflectivity for the neutron's given momentum change
+    """
+    R = R0
+    if Q > Qc:
+        tmp = (Q - m * Qc) / W
+        if tmp < 10:
+            R *= (1 - tanh(tmp)) * (1 - alpha * (Q - Qc)) / 2
+        else:
+            R = 0
```

Comment:
I'm surprised by this. Maybe this else clause shouldn't be here?

Reply:
When Q is much larger than mQc, reflectivity goes to zero. Basically it means no reflectivity for large incident angles.

Reply:
Aha, thank you, I think this fits what I was observing in some testing.